### DISCUSSION TITLE
[Feature]: properly sync the plugins on git changes

### DISCUSSION THREAD

**Discussion by kylo252** [LunarVim/LunarVim#1783] [Ideas]
### Feature motivation

We should have a robust way that detects changes or conflicts that could lead to Packer errors. Such changes can include: 

- list of core plugins being out-of-sync [added/removed entries] 
- list of user plugins being out-of-sync [added/removed entries] 
- missing `$LUNARVIM_CONFIG_DIR/plugins/packer_compiled.lua`
- corrupted `$LUNARVIM_CONFIG_DIR/plugins/packer_compiled.lua` [unlikely]

### Describe the expected solution

1. Check if the content of the core/user plugins list has changed
2. Perform a clean-up for Packer environment: 

	  ```bash
	  # semi psuedo-code
	  if [ -e $LUNARVIM_CONFIG_DIR/plugins/packer_compiled.lua ]; then
		    rm -rf $LUNARVIM_CONFIG_DIR/plugins/packer_compiled.lua
	  fi
	  # TODO: make sure the timeout is enough
	  lvim --headless +'autocmd User PackerComplete sleep 100m | qall' +PackerSync
	  # TODO: find a way for error handling
	  lvim --headless +'autocmd User PackerComplete sleep 100m | qall' +PackerCompile
	  ```

### Describe the alternatives you've considered

It might be possible to do with a git hook using https://github.com/facebook/watchman

Example:

https://github.com/git/git/blob/225bc32a989d7a22fa6addafd4ce7dcd04675dbf/templates/hooks--fsmonitor-watchman.sample#L1-L10

**Comment by abzcoding** [comment]
#1399

**Comment by ahmedkhalf** [comment]
I would like to suggest that if #1397 get's implemented then step 2 could easily be implemented in a cross-platform way. (see `lvim clean`)
